File: src/codility/java/sort/MaxProductOfThree.java
package codility.java.sort;

import java.util.Arrays;

public class MaxProductOfThree {
    public int solution(int[] A) {
        Arrays.sort(A);
        int max = A[0] * A[1] * A[A.length-1];
        return Math.max(max, A[A.length-1] * A[A.length-2] * A[A.length-3]);
    }
}

File: src/codility/kotlin/MaxProductOfThree.kt
package codility.kotlin

class MaxProductOfThree {

    fun solution(A: IntArray): Int {
        // write your code in Kotlin
        return A.sort().let {
            val N = A.size
            Math.max(A[N-3] * A[N -2] * A[N-1] , A[0] * A[1] * A[N-1])
        }

    }
}